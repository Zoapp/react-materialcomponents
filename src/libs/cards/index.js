/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
