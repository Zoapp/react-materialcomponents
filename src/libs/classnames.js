/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
