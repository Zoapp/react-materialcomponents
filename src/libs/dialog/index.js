/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
