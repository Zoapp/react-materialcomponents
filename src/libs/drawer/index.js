import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
