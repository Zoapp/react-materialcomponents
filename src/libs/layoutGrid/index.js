/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
