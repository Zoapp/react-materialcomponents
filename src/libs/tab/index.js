import Tabbar from "./tabbar";
import Tab from "./tab";

export { Tabbar, Tab };
