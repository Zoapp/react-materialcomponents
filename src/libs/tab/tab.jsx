/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Icon from "../components/icon";
import Rmdc from "../";

/*
mdc-tab
See:
https://material.io/components/web/catalog/tabs/
http://material-components-web.appspot.com/tabs.html

TODO:
- dynamic, actions
- Mixins
*/
const Tab = ({
  className, active, text, icon, href, ...props
}) => {
  let classes = "mdc-tab";
  if (active) {
    classes += " mdc-tab--active";
  }
  if (className) {
    classes += ` ${className}`;
  }
  let i = "";
  if (icon) {
    i = (<Icon name={icon} className="mdc-tab__icon" aria-hidden="true" />);
  }
  return Rmdc.render(<a className={classes} href={href} >{i}{text}</a>, props);
};

Tab.defaultProps = {
  className: null,

  active: false,
  text: null,
  icon: null,
  href: "#",
};

Tab.propTypes = {
// React component props
  className: PropTypes.string,

  active: PropTypes.bool,
  text: PropTypes.string,
  icon: PropTypes.string,
  href: PropTypes.string,
};

export default Tab;
