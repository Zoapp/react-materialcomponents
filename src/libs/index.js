/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
