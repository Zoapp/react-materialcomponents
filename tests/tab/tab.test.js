import React from "react";
import renderer from "react-test-renderer";
import { Tab } from "libs/tab";

describe("tab/Tab", () => {
  it("can have children", () => {
    const tree = renderer.create(<Tab>Children are here!</Tab>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be active", () => {
    const tree = renderer.create(<Tab active />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a text", () => {
    const tree = renderer.create(<Tab text="foobar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an icon", () => {
    const tree = renderer.create(<Tab icon="foobar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a link", () => {
    const tree = renderer.create(<Tab href="#" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an id", () => {
    const tree = renderer.create(<Tab tabId={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be colored", () => {
    const tree = renderer.create(<Tab color="pink" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
