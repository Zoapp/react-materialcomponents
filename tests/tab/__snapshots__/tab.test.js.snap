// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab/Tab can be active 1`] = `
<a
  className="mdc-tab mdc-tab--active"
  href={null}
  onClick={[Function]}
>
  
</a>
`;

exports[`tab/Tab can be colored 1`] = `
<a
  className="mdc-tab"
  href={null}
  onClick={[Function]}
  style={
    Object {
      "color": "pink",
    }
  }
>
  
</a>
`;

exports[`tab/Tab can have a link 1`] = `
<a
  className="mdc-tab"
  href="#"
  onClick={[Function]}
>
  
</a>
`;

exports[`tab/Tab can have a text 1`] = `
<a
  className="mdc-tab"
  href={null}
  onClick={[Function]}
>
  
  foobar
</a>
`;

exports[`tab/Tab can have an icon 1`] = `
<a
  className="mdc-tab"
  href={null}
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="mdc-tab__icon material-icons"
  >
    foobar
  </i>
</a>
`;

exports[`tab/Tab can have an id 1`] = `
<a
  className="mdc-tab"
  href={null}
  onClick={[Function]}
>
  
</a>
`;

exports[`tab/Tab can have children 1`] = `
<a
  className="mdc-tab"
  href={null}
  onClick={[Function]}
>
  
  Children are here!
</a>
`;
